frontend/server/src/main/java/org/pytorch/serve/http/TokenAuthorizationHandler.java

```diff
@@ -84,20 +108,208 @@ public static void setupTokenClass() {
     private void checkTokenAuthorization(FullHttpRequest req, String type) throws ModelException {
 
         try {
-            Method method =
-                    tokenClass.getMethod(
-                            "checkTokenAuthorization",
-                            io.netty.handler.codec.http.FullHttpRequest.class,
-                            String.class);
-            boolean result = (boolean) (method.invoke(tokenObject, req, type));
+            boolean result = tokenClass.checkTokenAuthorization(req, type);
             if (!result) {
                 throw new InvalidKeyException(
                         "Token Authentication failed. Token either incorrect, expired, or not provided correctly");
             }
-        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
-            e.printStackTrace();
+        } catch (Exception e) {
             throw new InvalidKeyException(
                     "Token Authentication failed. Token either incorrect, expired, or not provided correctly");
         }
     }
 }
+
+class Token {
+    private static String apiKey;
+    private static String managementKey;
+    private static String inferenceKey;
+    private static Instant managementExpirationTimeMinutes;
+    private static Instant inferenceExpirationTimeMinutes;
+    private static Double timeToExpirationMinutes;
```

Comment:
Lets get rid if the double bookkeeping here and clean out the code. Token is now part of TS so it can access ConfigManager. We can just read the time from there to have a single source of truth.